apps/meteor/tests/end-to-end/api/livechat/00-rooms.ts

```diff
@@ -64,6 +66,32 @@ const getSubscriptionForRoom = async (roomId: string, overrideCredential?: Crede
 	return subscription;
 };
 
+// For creating dangling subscriptions to rooms (which can't be created in usual conditions)
+const addSubscriptionToDb = async (userId: IUser['_id'], room: IOmnichannelRoom) => {
+	const subscription = {
+		open: false,
+		alert: false,
+		unread: 0,
+		userMentions: 0,
+		groupMentions: 0,
+		ts: room.ts,
+		rid: room._id,
+		name: room.name,
+		fname: room.fname,
+		...(room.customFields && { customFields: room.customFields }),
+		t: room.t,
+		u: {
+			_id: userId,
+			username: 'username',
+			name: 'name',
+		},
+	};
+	const connection = await MongoClient.connect(URL_MONGODB);
+
+	await connection.db().collection('rocketchat_subscription').insertOne(subscription);
+	await connection.close();
+};
+
```

Comment:
Let's not open this possibility to API tests, I'm concerned if we do this, we will start to insert things directly into the DB to make it easier to test, I'd say we should try to reproduce through the API only, if not possible we could create a unit test instead of open a DB connection here.

Reply:
Hmmm it's not possible to reproduce this only with the API since dangling subscriptions are only generated by race conditions (and we don't have any endpoint that would allow us to add a subscription to a closed room) -- or at least this could turn into a flaky test if we try to reproduce the race condition.

I thought of unit tests at first, but there seemed to be so much to  mock that made me believe that wouldn't be the right path. But I could try this out now

Reply:
Done! I've just replaced the end-to-end tests by unit tests